pkg/pod/entrypoint_lookup_impl_test.go

```diff
@@ -0,0 +1,180 @@
+)
+
+const (
+	username = "foo"
+	password = "bar"
+)
+
+type fakeHTTP struct {
+	reg http.Handler
+}
+
+func newfakeHTTP() fakeHTTP {
+	reg := registry.New()
+	return fakeHTTP{
+		reg: reg,
+	}
+}
+
+func (f *fakeHTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Request authentication for ping request
+	// For further reference see https://docs.docker.com/registry/spec/api/#api-version-check
+	if r.URL.Path == "/v2/" && r.Method == http.MethodGet {
+		w.Header().Add("WWW-Authenticate", "basic")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	// Check auth if we've fetching the image
+	if strings.HasPrefix(r.URL.Path, "/v2/task") && r.Method == "GET" {
+		u, p, ok := r.BasicAuth()
+		if !ok || username != u || password != p {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+	}
+	// Default to open
+	f.reg.ServeHTTP(w, r)
+}
+
+func TestGet(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ftp := newfakeHTTP()
+	s := httptest.NewServer(&ftp)
+	defer s.Close()
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		t.Errorf("Parsing url with an error: %v", err)
+	}
+
+	task := parse.MustParseTask(t, `metadata:name: test-create-image`)
+
+	ref, err := remotetest.CreateImageWithAnnotations(u.Host+"/task/test-create-image", remotetest.DefaultObjectAnnotationMapper, task)
+	if err != nil {
+		t.Errorf("uploading image failed unexpectedly with an error: %v", err)
+	}
+
+	imgRef, err := name.ParseReference(ref)
+	if err != nil {
+		t.Errorf("digest %s is not a valid reference: %v", ref, err)
+	}
+
+	for _, tc := range []struct {
+		name     string
+		username string
+		password string
+	}{{
+		name:     "correct secret",
+		username: username,
+		password: password,
+	}, {
+		name:     "unauthorized secret",
+		username: username,
+		password: "wrong password",
+	}, {
+		name:     "nonexistent secret",
+		username: "wrong username",
+		password: "wrong password",
+	}, {
+		name:     "empty secret",
+		username: "",
+		password: "",
+	}, {
+		name:     "no secret",
+		username: "",
+		password: "",
+	}} {
+		t.Run(tc.name, func(t *testing.T) {
+
+			client := fakeclient.NewSimpleClientset(&corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default",
+					Namespace: "ns",
+				},
+			}, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "secret",
+					Namespace: "ns",
+				},
+				Type: corev1.SecretTypeDockercfg,
+				Data: map[string][]byte{
+					corev1.DockerConfigKey: []byte(
+						fmt.Sprintf(`{%q: {"auth": %q}}`,
+							u.Host,
+							base64.StdEncoding.EncodeToString([]byte(tc.username+":"+tc.password)),
+						),
+					),
+				},
+			})
+
+			entrypointCache, err := NewEntrypointCache(client)
+			if err != nil {
+				t.Errorf("Creating entrypointCache with an error: %v", err)
+			}
+
+			imagePullSecrets := []corev1.LocalObjectReference{{Name: "secret"}}
+			if tc.name == "no secret" {
```

Comment:
i'd avoid switching on test case names - this is usually a good signal that the test case parameters need to be changed or you need to break up the tests.

e.g. instead of passing in username password, you can pass in secret refs.

Reply:
ah yes, thank you!